### ISSUE TITLE
[Feature] Add application icon

### ISSUE THREAD

**Issue by Stormersh** [ValveSoftware/steam-for-linux#8354]
#### Your system information

* Steam client version (build number or date): January 16
* Distribution (e.g. Ubuntu): Fedora 35
* Opted into Steam client beta?: [Yes]
* Have you checked for system updates?: [Yes]

#### Please describe your issue in as much detail as possible:

During the installation of a game, there's a prompt asking if you want a desktop shortcut and/or system application icon. If you fail to select it there's currently no way to add the latter after installation.

My suggestion would be to add one along "Add desktop shortcut" - This could help on issues like #7601, especially if it's possible to select several games.

![2022-01-20T23:25:52-03:00](https://user-images.githubusercontent.com/25964222/150454803-d7500c0c-db68-4276-901d-9203c829e94e.png)

#### Steps for reproducing this issue:

1. Start installation of a game
2. Forget to check "Create a system application icon" 
3. Finish installation
4. Right click the game, search for an option to create one
5. Cry?

**Comment by zee-mzha** [comment]
Any update on this? Can't reasonably be  that hard too add, just a new button calling some old code. I switched distros recently and I keep all my games on a separate drive, so for me it was a matter of simply showing steam where the folder is, however it doesn't create system icons/applications when you do this. I only managed to half fix it by manually adding a desktop shortcut to all my games then doing "mv ~/Desktop/* ~/.local/share/applications/", however this doesn't fix the issue of icons because steam doesn't correctly add the icons for those games and instead just uses the "steam" icon.

This is really annoying, but not the biggest deal in the world. It would be a lot more tolerable to fix this by hand if steam would at least tell me where its getting its icons from, it just auotmagically creates them into the ~/.local/share/icons/hicolor/ icon folders and you have no idea where the actual icons are stored. Conflicting answers all over the internet and all of them are wrong... Hell if it was a symlink at least I could figure out where the icons are and fix it myself.

**Comment by filvarga** [comment]
Looking for exactly the same thing. Hopefully this gets solved.

**Comment by kayomn** [comment]
This seems more like a bug than a feature request, as `.desktop` files added to desktop launchers during installation have the game icon

**Comment by filvarga** [comment]
They get only added if you check the checkbox. If you don't then you are out of luck.

**Comment by Lioncat6** [comment]
Yeah, this would be great to have as an option. Surely it would just be calling a few lines that would have otherwise run at the beginning of the install process.

**Comment by Syphdias** [comment]
Since I did not find this issue, I opened #982, because I only searched for adding "start menu" entry.

I thought I would drop by to add my current workaround if somebody finds this. But of course a menu entry for manage would really be appreciated!

#### My current workaround with "Sea of Thieves" as a example
- add desktop shortcut for the game:
  Right-click the game > Manage > Add desktop shortcut
- move the shortcut to the "start menu" folder: e.g.
  `mv ~/Desktop/Sea\ of\ Thieves.desktop ~/.local/share/applications/`

**Comment by Managor** [comment]
Commenting my interest in this feature so that it shows up with my `involves:` keyword searches.

**Comment by TheUnknownHack3r** [comment]
Commenting here as well to show interest.
For now, I guess one can make the shortcuts using the "Add desktop shortcut" button and move it to `/usr/share/applications` (This works for me on Arch Linux, KDE Plasma desktop)

**Comment by esdrasjnr** [comment]
+1

> I switched distros recently and I keep all my games on a separate drive [...]
Similar problem. Hope it gets solved soon, or someone find some workaround...

**Comment by Managor** [comment]
> Commenting here as well to show interest. For now, I guess one can make the shortcuts using the "Add desktop shortcut" button and move it to `/usr/share/applications` (This works for me on Arch Linux, KDE Plasma desktop)

`~/.local/share/applications/` exists so you don't have to mess with root permissions.

**Comment by jebez1** [comment]
After a re-installation of my Debian testing KDE, Steam Windows games on a separate partition, I need this feature.

The feature request dates from 2022, Valve doesn't seem to care...

**Comment by Managor** [comment]
There are hundreds of feature requests on this bugtracker and valve having their hands full with other projects, it's kinda expected for stuff like this gets neglected.

**Comment by jebez1** [comment]
@Managor
https://github.com/ValveSoftware/halflife/issues/3789

**Comment by FedericoHeichou** [comment]
If steam sets as icon "steam" instead of `steam_icon_<gameid>` is because you have already some broken links under `~/.local/share/icons/hicolor` called `steam_icon_<gameid>.png`.  
In my case I installed the game first with steam's snap, it didn't work so I uninstalled it and reinstalled from official steam repo.  
When I uninstalled the snap one, the links under the above folder are kept, but the target does not exists anymore.  
For example:

```
~/.local/share/icons/hicolor/192x192/apps/steam_icon_3224770.png -> /home/<username>/snap/steam/common/.local/share/icons/hicolor/192x192/apps/steam_icon_3224770.png
```

If you have this problem just delete them (replace the number with the id game, you can easly find it doing cat of .desktop generated entry, it is the number in `Exec`).
```bash
cd ~/.local/share/icons/hicolor
# Check the esistance of those broken links
find . -name steam_icon_3224770.png
# Delete them
find . -name steam_icon_3224770.png -exec rm {} \;
```
Now recreate the desktop entry from steam client and it will generates the `steam_icon_<gameid>.png` files correctly.  
Then you may need to "right click" and click "Permit execution".  
Finally you can move (or copy) it under `~/.local/share/applications`, so it will appear in the dash
